### DISCUSSION TITLE
气压计测量

### DISCUSSION THREAD

**Discussion by w00182024** [tszheichoi/awesome-sensor-logger#22] [General]
手机放到口袋场景下，APP不上报气压值，是怎么回事啊

**Comment by tszheichoi** [comment]
It depends on what kind of pocket you have placed your phone in. Normal trousers or jeans pockets should be sufficiently exposed to the ambient air pressure. As long as it is not pneumatically sealed, it should report the correct values.

**Comment by w00182024** [comment]
Thanks very much.  I continued some tests.
 I placed my phone on the desk, sometimes the data does not report. I think it is not pneumatically sealed. 
I exported the data to CSV format into a zip file, I don't know if the format expects the results.

**Reply by tszheichoi** [reply]
When you describe that the data does not report, are you referring to the sampling being intermittent, or does the barometer not report at all after some time? Is the app in the foreground? If it is in the background, do check you have granted the relevant permissions.

**Reply by w00182024** [reply]
I did a continuous sampling such as 15 minutes. The data reported sometimes, and sometimes not. 
The app is in the foregroud, and I have granted the relevant permissions.

**Reply by tszheichoi** [reply]
The only other thing I can think of is to ensure that "Keep Awake" is set to "On" in the "Recording & Workflow" settings. If the device is awake, the app is in the foreground, and permissions have been granted, then all the data sent from the sensor should be logged. If data is still missing, it is most likely that the device sensor has not sent any readings for some reason.
